src/App.js

```diff
@@ -1,14 +1,142 @@
+import Loading from './elements/Loading';
+import BackToTop from './elements/BackToTop';
+
+
 class App extends React.Component {
-  constructor(){
+  constructor() {
     super();
+    this.state = {
+      search: [],
+      searchTerm: "",
+      page: 2,
+      watchList: localStorage.getItem("watchList") ? localStorage.getItem("watchList") : [],
+      watchListSection: false
+    }
+    this.receiverSearch = this.receiverSearch.bind(this);
+    this.receiverWatchList = this.receiverWatchList.bind(this);
+    this.receiverDisplayWatchlist = this.receiverDisplayWatchlist.bind(this);
+    this.handleScroll = this.handleScroll.bind(this);
+  }
+
+  handleScroll() {
+    const windowHeight = "innerHeight" in window ? window.innerHeight : document.documentElement.offsetHeight;
+    const body = document.body;
+    const html = document.documentElement;
+    const docHeight = Math.max(body.scrollHeight, body.offsetHeight, html.clientHeight, html.scrollHeight, html.offsetHeight);
+    const windowBottom = windowHeight + window.pageYOffset;
+    const backToTop = body.querySelector("#back-to-top");
+
+    const loading = document.querySelector('#search-results__loading');
+    const OMBbAPIKey = "aba7af8e";
+    const query = 'http://www.omdbapi.com/?s=' + this.state.searchTerm + '&apikey=' + OMBbAPIKey + '&page=' + this.state.page;
+
+    // Back to top button
+    if (body.scrollTop > 300 || html.scrollTop > 300) {
+      backToTop.classList.add('visible');
+    } else {
+      backToTop.classList.remove('visible');
+    }
+
+    // Infinite scroll
+    if (windowBottom >= docHeight) {
+      fetch(query)
+        .then(response => {
+          loading.textContent = "Loading...";
+          loading.classList.add('show');
+          return response.json();
+        }
+        )
+        .then(movies => {
+          if (movies.Search) {
+            const self = this;
+            setTimeout(function () {
+              self.setState({
+                search: [...self.state.search, ...movies.Search],
+                page: self.state.page += 1
+              });
+              loading.classList.remove('show');
+            }, 500)
+          } else {
+            loading.textContent = "No more results";
+            loading.classList.add('show');
+          }
+        })
+        .catch(error => {
+          console.log(error);
+          loading.textContent = "There's been a problem fetching the results. Try again later..";
+          loading.classList.add('show');
+        });
+
+    }
+  }
+
+  componentDidMount() {
+    window.addEventListener("scroll", this.handleScroll);
+  }
+
+  receiverSearch(movies, searchTerm) {
+    this.setState({
+      search: movies,
+      searchTerm: searchTerm,
+      watchListSection: false
+    })
+  }
+
+  receiverDisplayWatchlist() {
+    this.setState({
+      watchListSection: true
+    })
+  }
+
+  receiverWatchList(articleTitle, action) {
+    if (action === "addToWatchList") {
```

Comment:
I would suggest implementing this as two functions. One will do `addToWatchList` and another `deleteFromWatchList`

Reply:
Keeping function to a single purpose will keep manageable and prevent them from becoming too large and doing too many things